doc/packages/build-support.md

```diff
@@ -0,0 +1,85 @@
+# Build Support {#sec-build-support}
+
+## `pkgs.substitute` {#pkgs-substitute}
+
+`pkgs.substitute` is a wrapper around [the `substitute` function](#fun-substitute) provided by the
+stdenv. It replaces strings in `src` as specified by the `substitutions` argument.
+
+Whereas in a build script, one might call `substitute $infile $outfile --replace-fail @foo@
+${foopkg}/bin/foo`, one might accomplish this with `pkgs.substitute` as such:
+
+```nix
+{ substitute, foopkg }:
+substitute {
+  src = ./sourcefile.txt;
+  substitutions = [
+    "--replace"
+    "@foo@"
+    "${foopkg}/bin/foo"
+  ];
+}
+```
+
+## `pkgs.substituteAll` {#pkgs-substituteall}
```

Comment:
we need a clear warning that uppercase vars and illegal identifiers (e.g. names with hyphens) are ignored

Reply:
Agreed! Will fix soon.

Reply:
How's that?

Reply:
Maybe it should be restated in the affirmative form?

```
As this uses the `substituteAll` function, its limitations also apply here,
meaning that variables must:

- start with a lowercase letter
- only contain alphanumeric characters or `_`
```

Reply:
Yes, I agree! updated.